src/components/dataset/query/QueryView.jsx

```diff
@@ -77,8 +82,21 @@ export class QueryView extends React.PureComponent {
   }
 
   componentDidUpdate(prevProps, prevState) {
-    const { dataset, dispatch, filterStatement, joinStatement, orderBy } = this.props;
-    const { selected } = this.state;
+    const { dataset, dispatch, filterStatement, joinStatement, orderBy, profile } = this.props;
+    const { selected, canLink } = this.state;
+
+    if (
+      this.hasDataset() &&
+      dataset.defaultProfileId &&
+      canLink === false &&
+      Object.keys(prevProps.profile).length === 0
+    ) {
+      dispatch(getBillingProfileById(dataset.defaultProfileId));
```

Comment:
It feels a little weird to be triggering the dispatch from the component. Could we have something that triggers this just based on the dataset being fetched successfully?